tests/testqPrefDisplay.cpp

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: GPL-2.0
+#include "testqPrefDisplay.h"
+
+#include "core/settings/qPrefDisplay.h"
+#include "core/pref.h"
+#include "core/qthelper.h"
+
+#include <QDate>
+
+void TestQPrefDisplay::initTestCase()
+{
+	QCoreApplication::setOrganizationName("Subsurface");
+	QCoreApplication::setOrganizationDomain("subsurface.hohndel.org");
+	QCoreApplication::setApplicationName("SubsurfaceTestQPrefDisplay");
+}
+
+void TestQPrefDisplay::test_struct_get()
+{
+	// Test struct pref -> get func.
+
+	auto display = qPrefDisplay::instance();
+
+	prefs.display_invalid_dives = true;
+	prefs.divelist_font = copy_qstring("comic");
+	prefs.font_size = 12.0;
+	prefs.show_developer = false;
+ 	prefs.theme = copy_qstring("myTheme");
+
+	QCOMPARE(display->display_invalid_dives(), prefs.display_invalid_dives);
+	QCOMPARE(display->divelist_font(), QString(prefs.divelist_font));
+	QCOMPARE(display->font_size(), prefs.font_size);
+	QCOMPARE(display->show_developer(), prefs.show_developer);
+	QCOMPARE(display->theme(), QString(prefs.theme));
+}
+
+void TestQPrefDisplay::test_set_struct()
+{
+	// Test set func -> struct pref
+
+	auto display = qPrefDisplay::instance();
+
+	display->set_display_invalid_dives(true);
+	display->set_divelist_font("comic");
+	display->set_font_size(12.0);
+	display->set_show_developer(false);
+ 	display->set_theme("myTheme");
+
+	QCOMPARE(prefs.display_invalid_dives, true);
+	QCOMPARE(prefs.divelist_font, "comic");
+	QCOMPARE(prefs.font_size, 12.0);
+	QCOMPARE(prefs.show_developer, false);
+	QCOMPARE(prefs.theme, "myTheme");
+}
+
+void TestQPrefDisplay::test_set_load_struct()
+{
+	// test set func -> load -> struct pref 
+
+	auto display = qPrefDisplay::instance();
+
+	display->set_display_invalid_dives(false);
+	display->set_divelist_font("helvitica");
```

Comment:
Perhaps I'm missing a play on words - but the original font is called "Helvetica" (i.e. hailing from Switzerland).

Reply:
good catch, except this is a test case so I could have called it (nearly) anything.

Reply:
I have changed it, in the PR dealing with qPrefDisplay #1491 (as stated in the PR this one depends on #1491, which is why you see those commits here).